File: Verifying an Alien Dictionary.java
class Solution {
    public boolean isAlienSorted(String[] words, String order) {
        boolean isSorted = true;
        for (int wordsIndex = 0; wordsIndex < words.length-1; wordsIndex++) {
            for (int charIndex = 0; charIndex < words[wordsIndex].length(); charIndex++) {  
                int word1Index = order.indexOf(words[wordsIndex].charAt(charIndex));
                int word2Index = order.indexOf(words[wordsIndex+1].charAt(charIndex));
                int word1Length = words[wordsIndex].length()-1;
                int word2Length = words[wordsIndex+1].length()-1;
                
                if (word1Index < word2Index) {
                    break;
                }
                
                if (word1Index > word2Index || (charIndex == word2Length && word1Length != word2Length)) {
                    isSorted = false;
                    break;
                }                       
            }
            
            if (!isSorted) {
                break;
            }
        }
        
        return isSorted;
    }
}

File: Verifying an Alien Dictionary.cpp
class Solution {
public:
    bool isAlienSorted(vector<string>& words, string order) {
        unordered_map<char, int> map;
        for (int i = 0; i < order.size(); i++) {
            map[order[i]] = i;
        }
        for (int i = 0; i < words.size() - 1; i++) {
            string word1 = words[i], word2 = words[i + 1];
            int j = 0, k = 0;
            while (j < word1.size() && k < word2.size()) {
                if (map[word1[j]] < map[word2[k]]) {
                    break;
                } else if (map[word1[j]] > map[word2[k]]) {
                    return false;
                }
                j++;
                k++;
            }
            if (j == word1.size() && k < word2.size()) {
                continue;
            }
            if (j < word1.size() && k == word2.size()) {
                return false;
            }
        }
        return true;
    }
};
